### PR TITLE
Ddp 5079 tb new langs

### PR DESCRIPTION
**Pull request by yufengwng**
## Context

Adding translations for 6 new languages for TestBoston. These are most of the changes needed. There's still a handful of things left to do, which I have documented in the ticket.

**TO DEPLOY TO `DEV` ENV:**

1. Run the following to push up Auth0 changes:
    ```
    $ ./render.sh v1 dev tenant
    $ a0deploy deploy -c output-config/testboston-config.json -i tenants/testboston/tenant.yaml
    ```
2. Render testboston configs and point to database:
    ```
    $ ./render.sh v1 dev testboston
    ```
3. Run the following study-builder tasks:
    1. `UpdateStudyLanguages`
    2. `UpdateActivityBaseSettings`
    3. `UpdateTemplatesInPlace`

## Checklist

- [x] I have labeled the type of changes involved using the `C-*` labels.
- [x] I have assessed potential risks and labeled using the `R-*` labels.
- [x] I have considered error handling and alerts, and added `L-*` labels as needed.
- [x] I have considered security and privacy, and added `I-*` labels as needed
- [x] I have analyzed my changes for stability, fault tolerance, graceful degradation, performance bottlenecks and written a brief summary in this PR.

## FUD Score

_Overall, how are you feeling about these changes?_

- [x] :relaxed: All good, business as usual!
- [ ] :sweat_smile: There might be some issues here
- [ ] :scream: I'm sweaty and nervous

## How do we demo these changes?

_How does one observe these changes in a deployed system? Note that **user visible** encompasses many personas--not just patients and study staff, but also ops duty, your fellow devs, compliance, etc._

- [ ] They are user-visible in dev as a regular user journey and require no additional instructions.
- [x] Getting dev into a state where this is user-visible requires some tech fiddling. I have documented these steps in the related ticket.
- [ ] Requires other features before it's human visible. I have documented the blocking issues in jira.
- [ ] I have no idea how to demo this. Please help me!

## Testing

- [ ] I have written automated positive tests
- [ ] I have written automated negative tests
- [x] I have written zero automated tests but have poked around locally to verify proper functionality
- [ ] The jira ticket has acceptance criteria and QA has the needed information to test changes

## Release

- [ ] These changes require no special release procedures--just code!
- [x] Releasing these changes requires special handling and I have documented the special procedures in the release plan document

### CODE REVIEW

File: pepper-apis/src/main/java/org/broadinstitute/ddp/db/dao/StudyLanguageDao.java [hunk 1]
```diff
@@ -43,6 +43,10 @@ default long insert(String studyGuid, String languageCode, String name) {
         return getStudyLanguageSql().insert(studyDto.getId(), languageCodeId, name);
     }
 
+    default void update(long umbrellaStudyId, long languageCodeId, String name) {
+        DBUtils.checkUpdate(1, getStudyLanguageSql().update(umbrellaStudyId, languageCodeId, name));
```
Comment (yufengwng) [inline pepper-apis/src/main/java/org/broadinstitute/ddp/db/dao/StudyLanguageDao.java:47]: These changes allow updating study languages in-place. See `UpdateStudyLanguages` task in study-builder.

File: study-builder/src/main/java/org/broadinstitute/ddp/studybuilder/EmailBuilder.java [hunk 2]
```diff
@@ -71,8 +72,8 @@ private void createForConfigs(List<Config> emailCfgs) {
                         email.getKey(), email.getTemplateId(), email.getVersionId());
             }
         }
-        Map<String, String> mappings = emails.stream()
-                .collect(Collectors.toMap(EmailInfo::getKey, EmailInfo::getTemplateId));
+        Map<String, String> mappings = new TreeMap<>();
```
Comment (yufengwng) [inline study-builder/src/main/java/org/broadinstitute/ddp/studybuilder/EmailBuilder.java:75]: I want the output to be sorted, so switched to a tree map here.

File: study-builder/studies/testboston/address.conf [hunk 3]
```diff
@@ -9,12 +9,24 @@
   "translatedNames": [
     { "language": "en", "text": ${i18n.en.address.name} },
     { "language": "es", "text": ${i18n.es.address.name} },
-    { "language": "ht", "text": ${i18n.ht.address.name} }
```
Comment (yufengwng) [inline study-builder/studies/testboston/address.conf:12]: Sorry about the 100+ files. Given how things are now, I had to touch almost every file in testboston in order to add the new translations into templates.

File: study-builder/tenants/testboston/pages/login.html [hunk 4]
```diff
@@ -20,7 +20,9 @@
 <![endif]-->
 
 <script src="https://cdn.auth0.com/js/lock/11.5.2/lock.min.js"></script>
+<script src="##BASE_URL##/assets/auth0-login-ar.js"></script>
 <script src="##BASE_URL##/assets/auth0-login-ht.js"></script>
+<script src="##BASE_URL##/assets/auth0-login-pt.js"></script>
```
Comment (yufengwng) [inline study-builder/tenants/testboston/pages/login.html:25]: These are hosted on the frontend app. See PR https://github.com/broadinstitute/ddp-angular/pull/317.

**Review by ssettipalli — APPROVED** [review]
verified java code changes and looks good to me.
If translations look good in local testing, we should be good.

**Pull request merged**